contracts/pool/PoolFactory.sol

```diff
@@ -2,22 +2,26 @@
 pragma solidity ^0.8.20;
 
 import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
 
 import {PoolFactoryLibrary} from "../library/PoolFactoryLibrary.sol";
 import {BlendedPoolFactoryLibrary} from "../library/BlendedPoolFactoryLibrary.sol";
 
 import {IHeliosGlobals} from "../interfaces/IHeliosGlobals.sol";
 import {IPoolFactory} from "../interfaces/IPoolFactory.sol";
-import {PoolErrors} from "./PoolErrors.sol";
+import {PoolErrors} from "./base/PoolErrors.sol";
 
 /// @title Factory for Pool creation
 /// @author Tigran Arakelyan
 contract PoolFactory is IPoolFactory, ReentrancyGuard, PoolErrors {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
     IHeliosGlobals public immutable override globals; // A HeliosGlobals instance
 
     address public blendedPool; // Address of Blended Pool
+    EnumerableSet.AddressSet private poolSet; // EnumerableSet of pools
+
     mapping(string => address) public pools; // Map to reference Pools corresponding to their respective indices.
```

Comment:
Storing strings on-chain is rather expensive thing to do, consider using bytes32 for pools names if I understood everything correctly.
So your pool name is keccak256("POOL NAME")
```suggestion
    mapping(bytes32 => address) public pools; // Map to reference Pools corresponding to their respective indices.
```